app/services/sponsored-content.js

```diff
@@ -0,0 +1,54 @@
+import Service, { inject as service } from '@ember/service';
+import { computed } from '@ember/object';
+
+export default Service.extend({
+  fetch: service(),
+  logger: service(),
+
+  items: null,
+  item: computed('items', function () {
+    return this.getSponsoredItem(this.items);
+  }),
+
+  fetchData() {
+    const url = this.fetch.getServiceUrl('wiki-recommendations', '/sponsored-articles');
+
+    if (!this.items) {
+      this.fetch.fetchAndParseResponse(url)
+        .then(data => this.set('items', data))
+        .catch((error) => {
+          this.logger.error(error.message);
+        });
+    }
+  },
+
+  getSponsoredItem(sponsoredContent = []) {
+    const randomNumber = Math.random();
+    const applicableContent = this.getApplicableContent(sponsoredContent);
+    const sumOfWeights = this.getWeightsSum(applicableContent);
+    const ranges = this.getMaxRanges(applicableContent, sumOfWeights);
+    const applicableRanges = ranges.filter(el => el >= randomNumber);
+    const firstApplicableIndex = applicableContent.length - applicableRanges.length;
+
+    return applicableContent[firstApplicableIndex];
+  },
+
+  getApplicableContent(sponsoredContent) {
+    // todo move to geo service
```

Comment:
add this fella https://github.com/Wikia/mobile-wiki/pull/1303/files#diff-1588ed5fc1201f6117a3586b0e5affd7R1